api/kv_test.go

```diff
@@ -445,3 +446,93 @@ func TestClient_AcquireRelease(t *testing.T) {
 		t.Fatalf("unexpected value: %#v", meta)
 	}
 }
+
+func TestClient_Txn(t *testing.T) {
+	t.Parallel()
+	c, s := makeClient(t)
+	defer s.Stop()
+
+	session := c.Session()
+	kv := c.KV()
+
+	// Make a session.
+	id, _, err := session.CreateNoChecks(nil, nil)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	defer session.Destroy(id, nil)
+
+	// Acquire and get the key via a transaction, but don't supply a valid
+	// session.
+	key := testKey()
+	value := []byte("test")
+	txn := KVTxn{
+		KVTxnOp{
+			Op:    KVLock,
+			Key:   key,
+			Value: value,
+		},
+		KVTxnOp{
+			Op:  KVGet,
+			Key: key,
+		},
+	}
+	ok, ret, _, err := kv.Txn(&txn, nil)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	} else if ok {
+		t.Fatalf("transaction should have failed")
+	}
```

Comment:
Basically the previous ~20LOC but added to the function comment block for `Txn`.